apps/browser/qml/pages/components/TabView.qml

```diff
@@ -40,131 +34,230 @@ SilicaGridView {
+
+        onCurrentIndexChanged: {
+            if (_remorsePopup) {
+                _remorsePopup.trigger()
+            }
+            privateMode = currentIndex !== 0
+        }
+        Component.onCompleted: {
+            persistentIcon.updateGrubImage()
+            privateIcon.updateGrubImage()
+            currentIndex = privateMode ? 1 : 0
         }
-    ]
 
-    ConfigurationValue {
-        id: showCloseAllAction
-        key: "/apps/sailfish-browser/settings/show_close_all"
-        defaultValue: true
-    }
+        Component {
+            id: persistentTabView
 
-    PullDownMenu {
-        id: pullDownMenu
-
-        flickable: tabView
-
-        MenuItem {
-            text: tabView.privateMode ?
-                    //: Menu item switching back to normal browser
-                    //% "Normal browsing"
-                    qsTrId("sailfish_browser-me-normal_browsing") :
-                    //: Menu item switching to private browser
-                    //% "Private browsing"
-                    qsTrId("sailfish_browser-me-private_browsing")
-            onDelayedClick: {
-                if (remorsePopup) {
-                    remorsePopup.trigger()
+            Private.TabItem {
+                allowDeletion: false
+                flickable: _persistentTabView
+
+                TabGridView {
+                    id: _persistentTabView
+                    privateMode: false
+                    portrait: root.portrait
+                    model: webView.persistentTabModel
+                    header: Item {
+                        width: 1
+                        height: tabs.tabBarHeight
+                    }
+
+                    onHide: root.hide()
+                    onEnterNewTabUrl: root.enterNewTabUrl()
+                    onActivateTab: root.activateTab(index)
+                    onCloseTab: root.closeTab(index)
+                    onCloseAll: root.closeAll()
+                    onCloseAllCanceled: root.closeAllCanceled()
+                    onCloseAllPending: root.closeAllPending()
+                    onContentYChanged: _changeMenuApparentHeight(contentY)
                 }
+                onIsCurrentItemChanged: {
+                    if (isCurrentItem) {
+                        _remorsePopup = Qt.binding(function() { return _persistentTabView.remorsePopup })
+
+                    }
+                }
+            }
+        }
+        Component {
+            id: privateTabView
+            Private.TabItem {
+                allowDeletion: false
+                flickable: _privateTabView
 
-                tabView.privateMode = !tabView.privateMode
+                TabGridView {
+                    id: _privateTabView
+                    privateMode: true
+                    portrait: root.portrait
+                    model: webView.privateTabModel
+                    header: Item {
+                        width: 1
+                        height: tabs.tabBarHeight
```

Comment:
Similarly, the same extra space would need to be added here:
```suggestion
                        height: tabs.tabBarHeight + Theme.paddingLarge
```

Reply:
done